### PR TITLE
xds: let GRPC_XDS_EXPERIMENTAL_ROUTING default to true

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

**Comment by ejona86** [comment]
This cannot be enabled until we are confident with the implementation, which implies interop testing completed.

File: xds/src/main/java/io/grpc/xds/XdsClientImpl.java [hunk 1]
```diff
@@ -97,8 +98,9 @@
       "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
 
   // Mutable for testing.
-  static boolean enableExperimentalRouting = Boolean.parseBoolean(
-      System.getenv("GRPC_XDS_EXPERIMENTAL_ROUTING"));
+  static boolean enableExperimentalRouting =
+      Strings.isNullOrEmpty(System.getenv("GRPC_XDS_EXPERIMENTAL_ROUTING"))
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/XdsClientImpl.java:102]: Why `Strings.isNullOrEmpty`?

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsClientImpl.java:102]: What's the problem with it?

Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/XdsClientImpl.java:102]: It's redundant.

**Comment by dapengzhang0** [comment]
We don't need this per the xds meeting decision.

**Pull request closed without merging**